akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -61,11 +61,20 @@ object ActorMaterializer {
       system,
       materializerSettings,
       system.dispatchers,
-      context.actorOf(StreamSupervisor.props(materializerSettings, haveShutDown).withDispatcher(materializerSettings.dispatcher), StreamSupervisor.nextName()),
+      spawnStreamSupervisor(materializerSettings, context, haveShutDown),
       haveShutDown,
       FlowNames(system).name.copy(namePrefix))
   }
 
+  private def spawnStreamSupervisor(materializerSettings: ActorMaterializerSettings, context: ActorRefFactory, haveShutDown: AtomicBoolean) =
+    context match {
+      case s: ActorSystemImpl ⇒
```

Comment:
`ExtendedActorSystem` should be "enough"

Reply:
right :)